### PR TITLE
docs: add logo and favicon, update theme and colors

### PR DESCRIPTION
**Pull request by umarcor**
Similarly to https://github.com/SymbiFlow/symbiflow-docs/pull/519, this PR adds a logo and favicon to the docs, so it's easier to identify the tab/page.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -13,10 +13,15 @@
 # All configuration values have a default; values that are commented out
 # serve to show the default.
 
+from pathlib import Path
+
+ROOT = Path(__file__).resolve().parent
```
Comment (mithro) [inline docs/source/conf.py:18]: Do you need to do this path trick? Can't you just put them under `_static`? https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_favicon says the path should be relative the conf file?

Comment (umarcor) [reply docs/source/conf.py]: I updated the docs to use `_static`.

**Comment by umarcor** [comment]
Converting to a draft, until building with Python 3 is fixed. See #2415.

**Comment by umarcor** [comment]
This PR now updates the theme, the colors and the copyright year.

Ref:

- SymbiFlow/sphinx_symbiflow_theme#7

**Comment by umarcor** [comment]
Since #2415 was merged, this PR is now ready for review: https://symbiflow--2414.org.readthedocs.build/projects/arch-defs/en/2414/.

**Comment by umarcor** [comment]
Logo and favicon were updated to F4PGA, and a `s/SymbiFlow/F4PGA/` commit was added.

**Pull request merged**